workflows/prognostic_c48_run/runtime/steppers/prescriber.py

```diff
@@ -0,0 +1,196 @@
+from typing import Sequence, MutableMapping, Mapping, Hashable, Tuple, Optional
+import dataclasses
+from datetime import timedelta
+import logging
+import intake
+import cftime
+import xarray as xr
+from runtime.types import State, Diagnostics, Tendencies
+import fv3gfs.util
+from vcm.catalog import catalog as CATALOG
+from vcm.safe import get_variables
+
+
+logger = logging.getLogger(__name__)
+
+QuantityState = MutableMapping[Hashable, fv3gfs.util.Quantity]
+
+
+def get_timesteps(
+    init_time: cftime.DatetimeJulian, timestep_seconds: float, n_timesteps: int
+) -> Sequence[cftime.DatetimeJulian]:
+    """Get sequence of model timesteps
+    
+    Args
+        init_time (cftime.DatetimeJulian): model run start time
+        timestep_seconds (float): model timestep
+        n_timesteps: number of timesteps in model run
+    
+    Returns: Sequence of cftime.DatetimeJulian objects corresponding to ends
+        of model run timesteps
+    
+    """
+    return [
+        init_time + timedelta(seconds=timestep_seconds * i)
+        for i in range(1, n_timesteps + 1)
+    ]
+
+
+@dataclasses.dataclass
+class PrescriberConfig:
+    """Configuration for a prescriber object to set states in the model from an external source
+
+    Attributes:
+        dataset_key (str): location of the dataset that provides prescribe values;
+            the routine first will try `dataset_key` as a `vcm.catalog` key, and if
+            not present it will attempt to use it as a (local or remote) path to
+            a zarr dataset
+        variables (Sequence[str]): sequence of "standardized" ("_coarse" suffix removed)
+            variable names to prescribe
+        rename (Mapping[Hashable, Hashable]): mapping of "standardized" ("_coarse"
+            suffix removed) names in the external dataset to variable names desired
+            for the runfile
+        consolidated (bool): optional, whether desired dataset has consolidated metadata;
+            defaults to True
+
+    Example::
+
+        PrescriberConfig(
+            dataset_key="gs://vcm-ml-intermediate/2021-03-fine-res-surface-radiative-fluxes/fine-res-surface-radiative-fluxes.zarr",
+            variables=[
+                "override_for_time_adjusted_total_sky_downward_shortwave_flux_at_surface", 
+                "override_for_time_adjusted_total_sky_downward_longwave_flux_at_surface", 
+                "override_for_time_adjusted_total_sky_net_shortwave_flux_at_surface",
+            ]
+        )
+
+    """  # noqa
+
+    dataset_key: str
+    variables: Sequence[str]
+    rename: Optional[Mapping[Hashable, Hashable]] = None
+    consolidated: bool = True
+
+
+class Prescriber:
+
+    net_moistening = "net_moistening"
```

Comment:
does this get used anywhere?

Reply:
No, but it's a legacy requirement of the `Stepper` base class.